modules/rostests/apitests/advapi32/RegCreateKeyExW.c

```diff
@@ -0,0 +1,158 @@
+
+// Based on code from the following:
+// https://learn.microsoft.com/en-us/windows/win32/secauthz/creating-a-security-descriptor-for-a-new-object-in-c--
+
+#include <apitest.h>
+#include <stdio.h>
+
+#define WIN32_NO_STATUS
+#define _INC_WINDOWS
+#define COM_NO_WINDOWS_H
+#include <windef.h>
+#include <aclapi.h>
+#include <tchar.h>
+#include <debug.h>
+
+
+START_TEST(RegCreateKeyExW)
+{
+    HKEY hkey_main;
+    DWORD dwRes, dwDisposition;
+    PACL pACL = NULL;
+    PSECURITY_DESCRIPTOR pSD = NULL;
+    PSID pEveryoneSID = NULL, pAdminSID = NULL;
+    SID_IDENTIFIER_AUTHORITY SIDAuthWorld = {SECURITY_WORLD_SID_AUTHORITY};
+    SID_IDENTIFIER_AUTHORITY SIDAuthNT = {SECURITY_NT_AUTHORITY};
+    EXPLICIT_ACCESS ea[2];
+    SECURITY_ATTRIBUTES sa;
+    LONG lRes;
+    BOOL bRes;
+    LONG ErrorCode = 0;
+    HKEY hkSub = NULL;
+
+    /* If any of the test keys already exist, delete them to ensure proper testing*/
+    if (RegOpenKeyExW(HKEY_CURRENT_USER, L"mykey", 0, KEY_ALL_ACCESS, &hkey_main) == ERROR_SUCCESS)
+        RegDeleteKeyExW(HKEY_CURRENT_USER, L"mykey", KEY_ALL_ACCESS, 0);
+
+    if (RegOpenKeyExW(HKEY_CURRENT_USER, L"mykey1", 0, KEY_ALL_ACCESS, &hkey_main) == ERROR_SUCCESS)
+        RegDeleteKeyExW(HKEY_CURRENT_USER, L"mykey1", KEY_ALL_ACCESS, 0);
+
+    if (RegOpenKeyExW(HKEY_CURRENT_USER, L"mykey2", 0, KEY_ALL_ACCESS, &hkey_main) == ERROR_SUCCESS)
+        RegDeleteKeyExW(HKEY_CURRENT_USER, L"mykey2", KEY_ALL_ACCESS, 0);
+
+    /* Setup GetLastError to known value for tests */
+    SetLastError(0xdeadbeef);
+
+    // Create a well-known SID for the Everyone group.
+    bRes = AllocateAndInitializeSid(&SIDAuthWorld, 1,
+        SECURITY_WORLD_RID,
+        0, 0, 0, 0, 0, 0, 0,
+        &pEveryoneSID);
+    ok(bRes == TRUE, "AllocateAndInitializeSid Error %ld\n", GetLastError());
+    if (bRes != TRUE) 
+        goto Cleanup;
+
+    // Initialize an EXPLICIT_ACCESS structure for an ACE.
```

Comment:
You could simplify this to a single KEY_ALL_ACCESS for Everyone for the purpose of the test.

Reply:
But ea is used to create pCL and pCL is used to create pSD and pSD is used to create the sa which we need for changing of the nLength for our tests. Sorry, I don't see how to do this.